internal/raterun/runner.go

```diff
@@ -1,102 +1,117 @@
 package raterun
 
 import (
+	"context"
 	"errors"
 	"time"
 )
 
-// RunFunction function to be executed at specified rate
-type RunFunction func(rate time.Duration, t time.Time)
+// RunFunction function to be executed every period
+type RunFunction func(period time.Duration)
 
-type Rate struct {
-	Start time.Duration // when to start this rate
-	Rate  time.Duration // run in amount of Duration
+type Schedule struct {
+	StartDelay time.Duration
+	Frequency  time.Duration
 }
 
-type Runner interface {
-	// Terminate cancels scheduling for the next run
-	Terminate()
-	// RestartRate resets function calling back to first defined rate
-	RestartRate()
-	// run starts running the function at rates provided to the constructor
-	Run()
-}
-
-// New creates a new runner which varies in time according to given rates
-func New(fn RunFunction, rates []Rate) (*Instance, error) {
-	if len(rates) == 0 {
-		return nil, errors.New("empty rates")
+func New(fn RunFunction, schedules []Schedule) (*Runner, error) {
```

Comment:
nit: a comment here would help me, I think.

// New creates a new `Runner` that will execute `fn` as defined by the provided `schedules`.
// Each `Schedule` in `schedules` defines how often `fn` should be executed at any given period of time.